### DISCUSSION TITLE
Filter contact who bought a specific kind of products ?

### DISCUSSION THREAD

**Discussion by apcloic** [YetiForceCompany/YetiForceCRM#15201] [How to] [answered]
Hi All,

Let's say you are selling microprocessors and you want to **filter** contacts that have bought an AMD last year.

As relationship between Contacts and Products is M:M, you cannot reach related module's field directly using filter condition, only 1:M related fields are available here.

So the only way I know is to write a little php script that will check a custom field (created for that purpose) if both conditions are true (AMD+last year) and then I could use a simple filter based on this checkbox.

It's not very convenient because if I want to filter contacts that have bought Intel this year, I need to edit the script or create a new one, etc.

Does anyone knows a better solution or do I missed something ?

Regards

**Comment by mariuszkrzaczkowski** [comment, marked as answer]
ideally, if the conditions in the filters were to be expanded with the relations M: M.
But no one has ordered this yet.
Alternatively, add a Products relation on the contact and add a field with the type "Values from the related module", it's not perfect but it should work.

**Reply by apcloic** [reply]
Thanks Marius, I will stay with my scripts for now as filters in "Values from the related module" type only works with dropdown fields.

**Comment by mariuszkrzaczkowski** [comment]
Today a new filtering mechanism for related records has been added, check if it will solve your problems.
![image](https://user-images.githubusercontent.com/10198654/147763306-f6b74812-2b88-46a7-bdf3-1293c52579e7.png)

**Reply by apcloic** [reply]
Wow, it's really nice new feature ! 
I've quickly check "conditions for filtering related records" which seems to work perfectly, it would be very cool if we could also save this kind of custom view in the future :)

For "relationship as a custom column", I've noticed that extra column remains empty but maybe I didn't understand how to make it work correctly.

Anyway, thank you very much for this // all the best to YF Team for New Year's Eve 🥇

**Reply by mariuszkrzaczkowski** [reply]
in custom filters is also available

**Reply by mariuszkrzaczkowski** [reply]
the new column is only used for searching, maybe the extension to show data

**Comment by apcloic** [comment]
@mariuszkrzaczkowski 

I think there is a little issue with count function in list view when using related records filtering mechanism.

On the screenshot below, the filter is based on Documents module with condition title is not empty.
Count result should be 2 instead of 6, so it looks like it count the related records (indeed there is a total of 6 documents) instead of records (there is 2 filtered contacts)

![image](https://user-images.githubusercontent.com/2930560/182660695-3ff7a4d7-9f3e-4e68-a90b-71f1ca9b8c7b.png)

![image](https://user-images.githubusercontent.com/2930560/182660563-9340633f-9486-43b9-82bb-9a8e514bc8f4.png)

Regards,

**Comment by apcloic** [comment]
Hi, 
The following filter condition allow to find all the contacts who have at least one related document.
![image](https://user-images.githubusercontent.com/2930560/182663851-9616d033-d398-4de4-9686-7d7f80ed361d.png)

As an improvement, I think it could be very useful to be able to also filter records who have no related records for the selected module.
Example : customers whose never bought any product, customers without any related email, etc.
